### ISSUE TITLE
IDE0079 false positive for IDE0032 in VB

### ISSUE THREAD

**Issue by doesicool** [dotnet/roslyn#84279]
> I know VB.NET is more or less dead language, but this issue is still not solved for it. When IDE0032 is supressed per attribute i get a IDE0079 false positive (TestClass2). See test code below:
> ```vb
> Imports System.Diagnostics.CodeAnalysis
> 
> Namespace Test.IDE0079.VB
> 	Public Class TestClass1
> 		' Correct warning IDE0032: Use auto property
> 		Private ReadOnly _Prop1 As Integer
> 		Public ReadOnly Property Prop1 As Integer
> 			Get
> 				Return _Prop1
> 			End Get
> 		End Property
> 	End Class
> 
> 	Public Class TestClass2
> 		' Suppress IDE0032 per attribute
> 		' IDE0079 false positiv
> 		<SuppressMessage("Style", "IDE0032:Use auto property")>
> 		Private ReadOnly _Prop1 As Integer
> 		Public ReadOnly Property Prop1 As Integer
> 			Get
> 				Return _Prop1
> 			End Get
> 		End Property
> 	End Class
> 
> 	Public Class TestClass3
> 		' Suppress IDE0032 per preprocessor directive
> 		' No IDE0079 false positiv
> #Disable Warning IDE0032 ' Use auto property
> 		Private ReadOnly _Prop1 As Integer
> #Enable Warning IDE0032 ' Use auto property
> 		Public ReadOnly Property Prop1 As Integer
> 			Get
> 				Return _Prop1
> 			End Get
> 		End Property
> 	End Class
> End Namespace
> ```
> (Additional Information: Tested with Visual Studio 2026 Version 18.7.2 in an .NET 8 class library with &lt;AnalysisLevel&gt; = preview-all.)
> 
> So please either mark issue with C# label or better reopen as unsolved for VB.NET. 

 _Originally posted by @doesicool in [#55529](https://github.com/dotnet/roslyn/issues/55529#issuecomment-4797428101)_

**Comment by CyrusNajmabadi** [comment]
This is a limitation of the supression system.  Because the attribute is there (regardless of what attribute it is), there is no reporting that this can be an auto prop.  That's because vb cannot change it to an auto prop, while preserving that attribute on the field.  

Because of that, the suppression *appears* unnecessary.  Because it's saying to suppress something that isn't being reported.

This is a known problem that no known solution exists for.

**Comment by doesicool** [comment]
I'm not sure if i fully understand. Because why does it correctly work then in C#?

**Comment by CyrusNajmabadi** [comment]
>Because why does it correctly work then in C#?

Because C# has a way to support arbitrary attributes when moving to an auto-prop (it uses `[field: OriginalAttr(...)]`).  So we *still* report the diagnostic, even with the `[Suppress...` attribute on it.  Thus making it seem as if the supression attribute cannot be removed.

VB only supports `<assembly:` and `<module:`, and thus the analyzer in the VB case bails out earlier as not being able to convert this.

**Comment by doesicool** [comment]
All right. So should i leave this issue open or close as not planned? I guess it will never be solved due to lack of support for the mentioned attributes.
